File: wlr/SPX.Paths/Walk.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SPX.Core;

namespace SPX.Paths
{
    class Walk
    {
        private static List<Point2>[] s_walks = new List<Point2>[1000];
        private static int m_index = 0;

        public static List<Point2> Copy(List<Point2> walk)
        {
            if (s_walks[0] == null)
            {
                for (int ii = 0; ii < s_walks.Count(); ii++)
                {
                    s_walks[ii] = new List<Point2>();
                }
            }
            m_index = (m_index + 1) % s_walks.Count();
            s_walks[m_index].Clear();
            foreach (var point in walk)
            {
                s_walks[m_index].Add(point);
            }
            return s_walks[m_index];
        }
    }
}

File: proj/src/spx/paths/Walk.java
package spx.paths;

import spx.core.Point2;

import java.util.ArrayList;
import java.util.List;

class Walk {
    private static ArrayList<List<Point2>> __walks = new ArrayList<List<Point2>>();
    private static int _index = 0;
    private static final int __walkCount = 1000;

    public static List<Point2> copy(List<Point2> walk) {
        if (__walks.size() == 0) {
            for (int ii = 0; ii < __walkCount; ii++) {
                __walks.add(new ArrayList<Point2>());
            }
        }
        _index = (_index + 1) % __walks.size();
        __walks.get(_index).clear();
        for (Point2 point : walk) {
            __walks.get(_index).add(point);
        }
        return __walks.get(_index);
    }
}
